realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SyncedRealmTests.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.SyncTestUtils.Companion.createTestUser
+import io.realm.entities.AllJavaTypes
+import io.realm.entities.AllTypes
+import org.junit.*
+import org.junit.Assert.*
+import org.junit.runner.RunWith
+import java.io.File
+import java.io.IOException
+
+/**
+ * Testing sync specific methods on [Realm].
+ */
+@RunWith(AndroidJUnit4::class)
+class SyncedRealmTests {
```

Comment:
There was a FIXME in `KotlinSyncedRealmTests`to merge it into this one once we migrated to Kotlin. Might as well be part of this PR.

Reply:
Done.